src/Auth0.php

```diff
@@ -412,38 +412,60 @@ public function __construct(array $config)
     public function login($state = null, $connection = null, array $additionalParams = [])
     {
         $params = [];
-        if ($this->audience) {
-            $params['audience'] = $this->audience;
-        }
 
-        if ($this->scope) {
-            $params['scope'] = $this->scope;
+        if ($state) {
+            $params['state'] = $state;
         }
 
-        if ($state === null) {
-            $state = $this->stateHandler->issue();
-        } else {
-            $this->stateHandler->store($state);
+        if ($connection) {
+            $params['connection'] = $connection;
         }
 
-        $params['response_mode'] = $this->responseMode;
-
         if (! empty($additionalParams) && is_array($additionalParams)) {
             $params = array_replace($params, $additionalParams);
         }
 
-        $url = $this->authentication->get_authorize_link(
-            $this->responseType,
-            $this->redirectUri,
-            $connection,
-            $state,
-            $params
-        );
+        $login_url = $this->getLoginUrl($params);
 
-        header('Location: '.$url);
+        header('Location: '.$login_url);
         exit;
     }
 
+    /**
+     * Build the login URL.
+     *
+     * @param array $params Array of authorize parameters to use.
+     *
+     * @return string
+     */
+    public function getLoginUrl(array $params = [])
```

Comment:
Does this method need to be public?

Reply:
Yes. Solves a problem in frameworks that have their own session and redirect handling (see auth0/laravel-auth0#135)